traffic_ops/install/bin/convert_profile/convert_profile.go

```diff
@@ -0,0 +1,293 @@
+/**
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Convert a Traffic Control Trafficserver Mid/Edge Cache Profile to a newer version
+ *
+ */
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type InputConfigParams struct {
+	InProfile string
+	OutFile   string
+	Rules     string
+	Force     bool
+}
+
+// TrafficOps Profile Parsing
+type Profile struct {
+	Parameters  []Parameter `json:"parameters"`
+	Description ProfileDesc `json:"profile"`
+}
+
+type Parameter struct {
+	Name       string `json:"name"`
+	ConfigFile string `json:"config_file"`
+	Value      string `json:"value"`
+}
+
+type ProfileDesc struct {
+	Description string `json:"description"`
+	Name        string `json:"name"`
+	Type        string `json:"type"`
+}
+
+// ConversionPolicy Parsing
+type ConversionPolicy struct {
+	ValidateParameters []Parameter      `json:"validate_parameters"`
+	ReplaceName        ReplaceRule      `json:"replace_name"`
+	ReplaceDescription ReplaceRule      `json:"replace_description"`
+	ConversionRules    []ConversionRule `json:"conversion_actions"`
+}
+
+type ReplaceRule struct {
+	Old string `json:"old"`
+	New string `json:"new"`
+}
+
+type ConversionRule struct {
+	MatchParameter Parameter `json:"match_parameter"`
+	NewName        string    `json:"new_name"`
+	NewConfigFile  string    `json:"new_config_file"`
+	NewValue       string    `json:"new_value"`
+	Action         string    `json:"action"`
+}
+
+func formatParam(p Parameter) string {
+	return fmt.Sprintf("{\"%s\", \"%s\", \"%s\"}", p.Name, p.ConfigFile, p.Value)
```

Comment:
Minor suggestion: this would be easier to read with backticks, ```
return fmt.Sprintf(`{"%s", "%s", "%s"}`, p.Name, p.ConfigFile, p.Value)
```

Reply:
thanks, i learned something new about Go today